### PR TITLE
AB#65917 AB#65916 AB#65915 Add rest tenure details views with logic

### PR DESCRIPTION
**Pull request by dawid-pawulski**
image

image

image

### CODE REVIEW

**Review by Carq — APPROVED** [review]

File: src/HE.Investment.AHP.Domain/HomeTypes/Entities/TenureDetailsSegmentEntity.cs [hunk 1]
```diff
@@ -135,13 +177,44 @@ public void ChangeExemptionJustification(MoreInformation? newValue)
...
+    {
+        const int exceed3PercentOfUnsoldShare = 80;
+        var result = CalculateProspectiveRentAsPercentageOfTheUnsoldShare(marketValue, prospectiveRent, initialSalePercentage);
+
+        var newValue = new Percentage(result);
+        SharedOwnershipRentAsPercentageOfTheUnsoldShare = _modificationTracker.Change(SharedOwnershipRentAsPercentageOfTheUnsoldShare, newValue);
+
+        if (result > exceed3PercentOfUnsoldShare)
+        {
+            IsExceeding3PercentOfUnsoldShare = true;
```
Comment (jjankowski87) [inline src/HE.Investment.AHP.Domain/HomeTypes/Entities/TenureDetailsSegmentEntity.cs:190]: it should be set to false in else (the same for 80% field)

**Review by jjankowski87 — APPROVED** [review]

File: src/HE.Investment.AHP.Domain/HomeTypes/CommandHandlers/SaveProspectiveRentCommandHandler.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    public SaveProspectiveRentCommandHandler(IHomeTypeRepository homeTypeRepository, ILogger<SaveProspectiveRentCommandHandler> logger)
+        : base(homeTypeRepository, logger)
+    {
+    }
+
+    protected override IReadOnlyCollection<HomeTypeSegmentType> SegmentTypes => new[] { HomeTypeSegmentType.TenureDetails };
+
+    protected override IEnumerable<Action<SaveProspectiveRentCommand, IHomeTypeEntity>> SaveActions => new[]
+    {
+        (SaveProspectiveRentCommand request, IHomeTypeEntity homeType) => homeType.TenureDetails.ChangeMarketValue(request.MarketValue),
```
Comment (Antek012) [inline src/HE.Investment.AHP.Domain/HomeTypes/CommandHandlers/SaveProspectiveRentCommandHandler.cs:19]: Shoudn't we have only single Agregate method call in handler?

File: src/HE.Investment.AHP.Domain/HomeTypes/Entities/TenureDetailsSegmentEntity.cs [hunk 3]
```diff
@@ -16,103 +14,147 @@ public class TenureDetailsSegmentEntity : IHomeTypeSegmentEntity
...
-    public bool IsExceeding80PercentOfMarketRent { get; private set; }
+    public InitialSale? InitialSale { get; private set; }
+
+    public ExpectedFirstTranche? ExpectedFirstTranche { get; private set; }
+
+    public Percentage? SharedOwnershipRentAsPercentageOfTheUnsoldShare { get; private set; }
+
+    public bool IsExceeding3PercentOfUnsoldShare { get; private set; }
+
+    public static decimal CalculateProspectiveRent(string? marketRent, string? prospectiveRent)
```
Comment (Antek012) [inline src/HE.Investment.AHP.Domain/HomeTypes/Entities/TenureDetailsSegmentEntity.cs:69]: Please add tests for methods like this.

**Review by Antek012 — APPROVED** [review]

File: src/HE.Investment.AHP.Domain/HomeTypes/Entities/TenureDetailsSegmentEntity.cs (hunk 1 above)
Comment (dawid-pawulski) [reply src/HE.Investment.AHP.Domain/HomeTypes/Entities/TenureDetailsSegmentEntity.cs]: done

File: src/HE.Investment.AHP.Domain/HomeTypes/CommandHandlers/SaveProspectiveRentCommandHandler.cs (hunk 2 above)
Comment (dawid-pawulski) [reply src/HE.Investment.AHP.Domain/HomeTypes/CommandHandlers/SaveProspectiveRentCommandHandler.cs:19]: I'll leave it as is for now, but it's something to consider and discuss

**Pull request merged**